lib/jekyll/commands/serve/servlet.rb

```diff
@@ -186,7 +186,7 @@ def validate_and_ensure_charset(_req, res)
           key = res.header.keys.grep(%r!content-type!i).first
           typ = res.header[key]
 
-          unless typ =~ %r!;\s*charset=!
+          unless %r!;\s*charset=!.match?(typ)
```

Comment:
These don't seem strictly necessary. Can we minimize the diff in this PR by not including these changes here?